app/routes/core.rb

```diff
@@ -41,6 +41,18 @@ class Core < Sinatra::Application
         cache_control :private
       end
 
+      before do
+        if session[:access_token] && current_user.guest?
+          begin
+            user = ::User.from_github(*Github.user_info(session[:access_token]))
+            login(user)
+          rescue Github::UnauthorizedUser => e
+            Bugsnag.notify(e, nil, request)
+            session[:access_token] = nil
```

Comment:
Similarly here, we'll want to rescue the unauthorized case and the network connectivity/API error case separately, notifying Bugsnag only when there's a network/API error, clearing the token only when unauthorized.